pkg/roll/run_update.go

```diff
@@ -684,7 +713,31 @@ func (u *update) countHealthy(id rcf.ID, checks map[types.NodeName]health.Result
 		ret.Unknown = ret.Desired - ret.Current
 	}
 
+	var eligibleNodes []types.NodeName
+	if rcFields.AllocationStrategy == rcf.DynamicStrategy {
+		fmt.Println("calling bro")
+		eligibleNodes, err = u.scheduler.EligibleNodes(rcFields.Manifest, rcFields.NodeSelector)
+		if err != nil {
+			return ret, util.Errorf("could not enumerate eligible nodes for %s: %s", rcFields.ID, err)
+		}
+	}
+
 	for _, pod := range currentPods {
+		if rcFields.AllocationStrategy == rcf.DynamicStrategy {
```

Comment:
nit: you could instead check for AllocationStrategy != Dynamic and continue to reduce the indentation. Kind of in the style of https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
I don't think that would reduce indentation and it would make the if block longer since I would have to include the section that counts Healthy

Reply:
You're right. This is the same loop I misread the end of. Ignore me